src/library/tactic/user_attribute.cpp

```diff
@@ -117,6 +160,16 @@ vm_obj attribute_fingerprint(vm_obj const & vm_n, vm_obj const & vm_s) {
     return mk_tactic_success(mk_vm_nat(h), s);
 }
 
+vm_obj caching_user_attribute_get_cache(vm_obj const & vm_attr, vm_obj const & vm_s) {
+    auto const & s = to_tactic_state(vm_s);
+    auto const & n = to_name(cfield(vm_attr, 0));
+    auto const & cache_handler = cfield(vm_attr, 2);
```

Comment:
I see, thanks!

On Mon, Sep 12, 2016 at 1:04 PM Leonardo de Moura <notifications@github.com>
wrote:

> In src/library/tactic/user_attribute.cpp
> <https://github.com/leanprover/lean/pull/1129#discussion_r78411740>:
>
> > @@ -117,6 +160,16 @@ vm_obj attribute_fingerprint(vm_obj const & vm_n, vm_obj const & vm_s) {
> >      return mk_tactic_success(mk_vm_nat(h), s);
> >  }
> >
> > +vm_obj caching_user_attribute_get_cache(vm_obj const & vm_attr, vm_obj const & vm_s) {
> > +    auto const & s = to_tactic_state(vm_s);
> > +    auto const & n = to_name(cfield(vm_attr, 0));
> > +    auto const & cache_handler = cfield(vm_attr, 2);
>
> The code generator erases all fields that are:
> 1) proofs (since they are irrelevant)
> 2) types (also computationally irrelevant).
> The following function returns this information:
> https://github.com/leanprover/lean/blob/lean3/src/library/util.h#L87
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/leanprover/lean/pull/1129/files/49972112801f40e569d15fe6355ecf68c5236afa#r78411740>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGqRpU9jYFZ23UmoVmAm14ZqODi7j0jks5qpYWlgaJpZM4J6JhE>
> .
>